lib/ec2imgutils/ec2setup.py

```diff
@@ -57,7 +57,15 @@ def clean_up(self):
     def create_security_group(self, vpc_id=None):
         if self.verbose:
             print('Creating temporary security group')
-        group_name = 'ec2uploadimg-%s' % (random.randint(1, 100))
+        group_name = ''
+        # Avoid name collisions
+        while not group_name:
+            group_name = 'ec2uploadimg-%s' % (random.randint(1, 100))
+            groups = self._connect().describe_security_groups()
```

Comment:
This will work and is an enhancement but introduces the chances for a race condition. May be better to do a while loop around a try catch on the security group creation.

Reply:
Good catch, thanks. Fixed